### PR TITLE
Switch cli to github.com/spf13/cobra

### PR DESCRIPTION
**Pull request by abhinavdahiya**
This allows us to use the already vendored cobra cli package, and drop gopkg.in/alecthomas/kingpin.v2.
This also splits cli into 3 different command groups: targets, destroy and version.
cobra allows us to add new subcommands and local flags without putting everything into single switch in main.go

/cc @wking

### CODE REVIEW

File: cmd/openshift-install/targets.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+	clusterCmd = &cobra.Command{
+		Use:   "cluster",
+		Short: "Create an OpenShift cluster",
+		Long:  "",
+		RunE:  runTargetCmd(&cluster.TerraformVariables{}, &kubeconfig.Admin{}, &cluster.Cluster{}, &metadata.Metadata{}),
+	}
+)
+
+func init() {
+	rootCmd.AddCommand(installConfigCmd)
```
Comment (wking) [inline cmd/openshift-install/targets.go]: I'd rather have these in `main.go` like [this][1].

[1]: https://github.com/kubernetes/kubernetes/blob/v1.12.1/pkg/kubectl/cmd/cmd.go#L404

File: cmd/openshift-install/main.go [hunk 2]
```diff
@@ -1,121 +1,51 @@
...
 
 var (
-	installConfigCommand   = kingpin.Command("install-config", "Generate the Install Config asset")
-	ignitionConfigsCommand = kingpin.Command("ignition-configs", "Generate the Ignition Config assets")
-	manifestsCommand       = kingpin.Command("manifests", "Generate the Kubernetes manifests")
-	clusterCommand         = kingpin.Command("cluster", "Create an OpenShift cluster")
-	versionCommand         = kingpin.Command("version", "Print version information and exit")
-
-	destroyCommand = kingpin.Command("destroy-cluster", "Destroy an OpenShift cluster")
+	version = "was not built correctly" // set in hack/build.sh
```
Comment (wking) [inline cmd/openshift-install/main.go]: This shoiuld probably move to `version.go`.

File: cmd/openshift-install/main.go [hunk 3]
```diff
@@ -1,121 +1,51 @@
...
 
-	version = "was not built correctly" // set in hack/build.sh
+	rootOpts struct {
+		dir      string
+		logLevel string
+	}
 )
 
+func init() {
+	rootCmd.PersistentFlags().StringVar(&rootOpts.dir, "dir", ".", "assets directory")
```
[the remark this answers is not in the document]
Comment (abhinavdahiya) [reply cmd/openshift-install/main.go]: sure

File: cmd/openshift-install/targets.go (hunk 1 above)
Comment (abhinavdahiya) [reply cmd/openshift-install/targets.go]: i think `kubectl` example is more useful if we have more complex commands. but i can add `new<cmd>` type funcs.

File: cmd/openshift-install/main.go (hunk 2 above)
Comment (abhinavdahiya) [reply cmd/openshift-install/main.go]: sure.

File: cmd/openshift-install/targets.go (hunk 1 above)
Comment (wking) [reply cmd/openshift-install/targets.go]: No I just meant "not in an `init()`" ;).

**Comment by abhinavdahiya** [comment]
@wking updated to remove `init()` funcs... does this look better?

**Pull request merged**